### ISSUE TITLE
Unable to destroy activity

### ISSUE THREAD

**Issue by Danny-James** [mapbox/mapbox-navigation-android#1060]
**Gradle**
```
    implementation('com.mapbox.mapboxsdk:mapbox-android-navigation-ui:0.15.0') {
        transitive = true
        changing = true
    }
```

The following error occurs when touching the `X` to exit the `NavigationView`.

**Error**
```
06-25 17:45:58.910 5464-5464/uk.co.ainscough.routeinfo E/AndroidRuntime: FATAL EXCEPTION: main
    Process: uk.co.ainscough.routeinfo, PID: 5464
    java.lang.RuntimeException: Unable to destroy activity {uk.co.ainscough.routeinfo/uk.co.ainscough.routeinfo.ActivityNavigation}: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapbox.services.android.navigation.ui.v5.voice.MapboxSpeechPlayer.onDestroy()' on a null object reference
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4603)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4621)
        at android.app.ActivityThread.-wrap5(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1757)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapbox.services.android.navigation.ui.v5.voice.MapboxSpeechPlayer.onDestroy()' on a null object reference
        at com.mapbox.services.android.navigation.ui.v5.voice.NavigationInstructionPlayer.onDestroy(NavigationInstructionPlayer.java:60)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.deactivateInstructionPlayer(NavigationViewModel.java:349)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.onDestroy(NavigationViewModel.java:94)
        at com.mapbox.services.android.navigation.ui.v5.NavigationView.shutdown(NavigationView.java:522)
        at com.mapbox.services.android.navigation.ui.v5.NavigationView.onDestroy(NavigationView.java:161)
        at uk.co.ainscough.routeinfo.ActivityNavigation.onDestroy(ActivityNavigation.java:138)
        at android.app.Activity.performDestroy(Activity.java:7462)
        at android.app.Instrumentation.callActivityOnDestroy(Instrumentation.java:1255)
        at android.app.ActivityThread.performDestroyActivity(ActivityThread.java:4590)
        at android.app.ActivityThread.handleDestroyActivity(ActivityThread.java:4621) 
        at android.app.ActivityThread.-wrap5(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1757) 
        at android.os.Handler.dispatchMessage(Handler.java:105) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6938) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374) 
```

**Error Location**
```java
 @Override
    protected void onDestroy() {
        super.onDestroy();
        navigationView.onDestroy();
    }
```

**Comment by danesfeder** [comment]
Hey @Danny-James this bug has been identified and is being addressed in #1054 - thanks for bringing it to our attention.  As a side question - what does your `NavigationRoute` request code look like?

**Comment by Danny-James** [comment]
@danesfeder I don't seem to be using `NavigationRoute`, I'm using the following;

```
MapboxMapMatching.Builder

NavigationViewOptions.Builder

NavigationViewOptions navigationViewOptions = setupOptions(directionsRoute);
navigationView.startNavigation(navigationViewOptions);
```

**Comment by danesfeder** [comment]
Ah right, forgot you were using `MapboxMapMatching.Builder` - this makes sense as well because we weren't exposing the `voiceLanguage` for `MapMatching` upstream.  This is also moving here > https://github.com/mapbox/mapbox-java/pull/847 

Thanks @Danny-James

**Comment by Danny-James** [comment]
@danesfeder 

```java
06-26 17:32:12.746 19549-19549/uk.co.ainscough.routeinfo E/AndroidRuntime: FATAL EXCEPTION: main
    Process: uk.co.ainscough.routeinfo, PID: 19549
    java.lang.NullPointerException: Attempt to invoke virtual method 'void com.mapbox.services.android.navigation.ui.v5.voice.MapboxSpeechPlayer.setMuted(boolean)' on a null object reference
        at com.mapbox.services.android.navigation.ui.v5.voice.NavigationInstructionPlayer.setMuted(NavigationInstructionPlayer.java:50)
        at com.mapbox.services.android.navigation.ui.v5.NavigationViewModel.setMuted(NavigationViewModel.java:100)
        at com.mapbox.services.android.navigation.ui.v5.instruction.InstructionView$6.onClick(InstructionView.java:593)
        at android.view.View.performClick(View.java:6891)
        at android.view.View$PerformClick.run(View.java:26083)
        at android.os.Handler.handleCallback(Handler.java:789)
        at android.os.Handler.dispatchMessage(Handler.java:98)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
06-26 17:32:12.913 19549-19713/uk.co.ainscough.routeinfo D/CrashlyticsCore: Session 5B326A630372-0001-4C5D-5335A1A0493D has fatal exception: true
    Session 5B326A630372-0001-4C5D-5335A1A0493D has non-fatal exceptions: false
```

Is this related? - happens when i attempt to mute the Navigation - I have noticed I don't seem to have any Voice Instructions on my last run of the Navigation

**Comment by danesfeder** [comment]
Yeah that's related @Danny-James, also fixed in #1054

**Comment by Danny-James** [comment]
@danesfeder Thanks!
